lib/facility.rb

```diff
@@ -1,14 +1,63 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+  :address,
+  :phone,
+  :services,
+  :registered_vehicles,
+  :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info_hash)
+    @name = info_hash[:name]
+    @address = info_hash[:address]
+    @phone = info_hash[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return if !@services.include?("Vehicle Registration")
+
+    vehicle.register
+    
+    case vehicle.plate_type
+    when :regular
+      @collected_fees += 100
+    when :antique
+      @collected_fees += 25
+    when :ev
+      @collected_fees += 200
+    end
+
+    @registered_vehicles << vehicle
+  end
+
+  def administer_written_test(registrant)
+    return false if !services.include?("Written Test")
+    return false if !registrant.permit?
+    return false if registrant.age < 16
+
+    registrant.update_license_data(:written)
```

Comment:
Great delegation to the class that is the best candidate to handle this logic! This is a great example of using encapsulation in your code. Same with delegating `register` to vehicle on line 25 :)